### PR TITLE
refactor(ui): add ui social connector filters

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add ui social connector filters
extract the hook utils file

image

image

## Linear Issue Reference

log-2437

## Testing

add ut case test locally
test locally UT case added

@logto-io/eng

### CODE REVIEW

File: packages/ui/src/hooks/utils.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+export const stateValidation = (state: string, connectorId: string) => {
+  const stateStorage = sessionStorage.getItem(`${storageKeyPrefix}:${connectorId}`);
+
+  return stateStorage === state;
+};
+
+export const storeState = (state: string, connectorId: string) => {
+  sessionStorage.setItem(`${storageKeyPrefix}:${connectorId}`, state);
+};
```
Comment (simeng-li) [inline packages/ui/src/hooks/utils.ts:59]: All the codes above are moved from the original useSocial hook file.

File: packages/ui/src/hooks/utils.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  if (!isNativeWebview()) {
+    for (const connector of socialConnectors) {
+      const { platform, target } = connector;
+
+      if (platform === 'Native') {
+        continue;
+      }
+
+      if (platform === 'Web' || !connectorMap.get(target)) {
```
Comment (wangsijie) [inline packages/ui/src/hooks/utils.ts:79]: nit: better add some comments, it takes me a bit time to understand.

**Review by wangsijie — APPROVED** [review]
LGTM

File: packages/ui/src/hooks/utils.ts (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/hooks/utils.ts:79]: good point

**Pull request merged**